flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BarrierBuffer.java

```diff
@@ -112,6 +113,8 @@
 	/** Flag to indicate whether we have drawn all available input. */
 	private boolean endOfStream;
 
+	private boolean isFinished;
```

Comment:
No comment for this field, I think this should be comment because it has a different behavior with `BufferTracker.java` about `finish`.